### PR TITLE
Adding chaincode API range query test and skip main

### PR DESCRIPTION
**Pull request by binhn**
## Description

Adding unit test on range query API. Adding unit test to skip the main module.
## Motivation and Context

Providing more unit test coverage
## How Has This Been Tested?

run `go test`
## Checklist:

- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [x] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: binhn@us.ibm.com

### CODE REVIEW

File: peer/main_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+package main
+
+import (
+	"testing"
+)
+
+func TestMain(t *testing.T) {
+	t.Skip("No needs to test main since it is covered by other tests")
+}
```
Comment (ghaskins) [inline peer/main_test.go:25]: Hi Binh,

I realize from your PR Description comment that the goal of skipping the tests was intentional, and thus justifies why this file is here.  What's not clear is why this is needed.  I am more curious than anything, but perhaps a comment in this file is a good idea as I suspect I won't be the only one wondering what this accomplishes.

File: core/chaincode/exectransaction_test.go [hunk 2]
```diff
@@ -1312,6 +1312,77 @@ func TestChaincodeQueryChaincodeWithSec(t *testing.T) {
...
+		t.Fail()
+		t.Logf("Error invoking <%s>: %s", chaincodeID, err)
+		GetChain(DefaultChain).Stop(ctxt, &pb.ChaincodeDeploymentSpec{ChaincodeSpec: spec})
+		closeListenerAndSleep(lis)
+		return
+	}
+	GetChain(DefaultChain).Stop(ctxt, &pb.ChaincodeDeploymentSpec{ChaincodeSpec: spec})
+	closeListenerAndSleep(lis)
+}
+
```
Comment (muralisrini) [inline core/chaincode/exectransaction_test.go:1385]: TestRangeQuery LGTM, thanks @binhn !

**Comment by binhn** [comment]
@ghaskins I'd like to have an indication that we don't need to test this directory/module, so that the coverage report doesn't look like this `?    github.com/hyperledger/fabric/peer  [no test files]`  but more like `ok     github.com/hyperledger/fabric/peer  0.008s  coverage: 0.0% of statements`.

It seems to mean the same thing, but if someone looks into that directory, he would see the test file and the intent. Let me known if you're aware of a better method.

**Comment by ghaskins** [comment]
Thanks Binh,

LGTM,

Reviewed-by: Greg Haskins gregory.haskins@gmail.com

**Pull request merged**